pysidt/sidt.py

```diff
@@ -317,19 +357,42 @@ def write_nodes(tree, file):
             p = None
         else:
             p = node.parent.name
+            
+        try:
+            json.dumps(node.rule)
+            rule = node.rule
+        except (TypeError, OverflowError):
+            rule = to_dict(node.rule) #will work on all rmgmolecule objects, new objects need this method implemented
+            try:
+                json.dumps(rule)
+            except:
+                print(rule)
+                raise ValueError 
+            
         nodesdict[node.name] = {
             "group": node.group.to_adjacency_list(),
-            "rule": node.rule,
+            "rule": rule,
             "parent": p,
             "children": [x.name for x in node.children],
             "name": node.name,
+            "depth": node.depth
         }
 
     with open(file, "w") as f:
         json.dump(nodesdict, f)
 
 
-def read_nodes(file):
+def read_nodes(file,class_dict=dict()):
```

Comment:
```suggestion
def read_nodes(file, class_dict=dict()):
```

Can you set default value to None and initialize it within the function